neo/rawio/openephysbinaryrawio.py

```diff
@@ -253,30 +252,55 @@ def _parse_header(self):
                     # 'states' was introduced in OpenEphys v0.6. For previous versions, events used 'channel_states'
                     if "states" in info or "channel_states" in info:
                         states = info["channel_states"] if "channel_states" in info else info["states"]
+
                         if states.size > 0:
                             timestamps = info["timestamps"]
                             labels = info["labels"]
-                            rising = np.where(states > 0)[0]
-                            falling = np.where(states < 0)[0]
 
-                            # infer durations
+                            # Identify unique channels based on state values
+                            channels = np.unique(np.abs(states))
+
+                            rising_indices = []
+                            falling_indices = []
+
+                            # all channels are packed into the same `states` array. 
+                            # So the states array includes positive and negative values for each channel:
+                            #  for example channel one rising would be +1 and channel one falling would be -1, 
+                            # channel two rising would be +2 and channel two falling would be -2, etc.
+                            # This is the case for sure for version >= 0.6.x.
+                            for channel in channels:
+                                # Find rising and falling edges for each channel
+                                rising = np.where(states == channel)[0]
+                                falling = np.where(states == -channel)[0]
+
+                                # Ensure each rising has a corresponding falling
+                                if rising.size > 0 and falling.size > 0:
+                                    if rising[0] > falling[0]:
+                                        falling = falling[1:]
+                                    if rising.size > falling.size:
+                                        rising = rising[:-1]
+
+                                    rising_indices.extend(rising)
+                                    falling_indices.extend(falling)
```

Comment:
@vigji maybe it's better to skip the extend in case `rising_indices != falling_indices` here? Otherwise an issue with one channel will result in all durations being lost!

We could also add a warning in case this happens for a certain channel. What do you think?

Reply:
I guess there is also a question of what the edge cases might be. Is there really a scenario where this can break down? The only option is the acquisition board to be missing rising/falling edges, but I am not sure at that point I would still use the data - it is possible that could have happened for other digital channels as well, and pairs of miss-detections could go unnoticed, in principle.

Have you seen or heard of any of those edge scenarios?

Reply:
One case could be if I accidentally stop a recording midstimulus (a mouse wakes up or the head-fixation apparatus comes loose). So maybe I want to save data, but I don't have a fall for the last "channel" of stim.  So I agree with Alessio!

Reply:
That would still work though because [here](https://github.com/iurillilab/python-neo/blob/fixing-ttl-multichan/neo/rawio/openephysbinaryrawio.py#L277-L281) we check if the first falling is before the first rising and viceversa.

Reply:
you raise a valid point @zm711 , but addressing that would require completely changing the way the interface represents digital data (currently, timestamp+duration of the ON state). Not sure this is the place for that. I agree that the current representation is not the most general, and I would be in favour of changing it, but I assume there might be a lot of code by now depending on it (or not - the fact that the bug addressed in this PR was still around lets me wondering how many people use this interface for multi-digital channel experiment logging).

@alejoe91 so you suggest to just add a warning, with no changes to the current logic in the end?

Reply:
He's saying add those lines on top of your PR to skip a channel if it doesn't fit those conditions.

And fair enough. We have to live with the code we have :)

Reply:
Should be gtg now!